packages/spectral/src/serverTypes/convertIntegration.ts

```diff
@@ -343,8 +351,10 @@ const codeNativeIntegrationComponent = (
               "The function that will be executed by the flow to return an HTTP response.",
           },
           perform: onTrigger,
-          onInstanceDeploy: onInstanceDeploy,
-          onInstanceDelete: onInstanceDelete,
+          onInstanceDeploy,
+          hasOnInstanceDeploy: !!onInstanceDeploy,
+          onInstanceDelete,
+          hasOnInstanceDelete: !!onInstanceDelete,
```

Comment:
When are these new `has*` properties used?

Reply:
They indicate the existence of these deploy/delete functions to the API so we can optimize later checks without having to consult the actual component code.